lib/Basic/Cache.cpp

```diff
@@ -78,8 +78,10 @@ void CacheImpl::setAndRetain(void *Key, void *Value, size_t Cost) {
   DefaultCacheKey CKey(Key, &DCache.CBs);
   auto Entry = DCache.Entries.find(CKey);
   if (Entry != DCache.Entries.end()) {
-    if (Entry->second == Value)
+    if (Entry->second == Value) {
+      DCache.CBs.keyDestroyCB(Key, nullptr);
```

Comment:
The documentation says we should release the original key, which is also how it behaves on macOS.  I think we should swap `Entry->first.Key` for the new key here.

Reply:
Done.